src/java/picard/analysis/CollectGcBiasMetrics.java

```diff
@@ -101,107 +83,75 @@
+    byte[] refBases;
+    private SAMFileHeader save_header;
 
     /** Stock main method. */
     public static void main(final String[] args) {
         System.exit(new CollectGcBiasMetrics().instanceMain(args));
     }
 
-    protected int doWork() {
-        IOUtil.assertFileIsReadable(REFERENCE_SEQUENCE);
-        IOUtil.assertFileIsReadable(INPUT);
-        IOUtil.assertFileIsWritable(OUTPUT);
+    @Override
+    protected void setup(final SAMFileHeader header, final File samFile) {
         IOUtil.assertFileIsWritable(CHART_OUTPUT);
         if (SUMMARY_OUTPUT != null) IOUtil.assertFileIsWritable(SUMMARY_OUTPUT);
-
-        // Histograms to track the number of windows at each GC, and the number of read starts
-        // at windows of each GC
-        final int[] windowsByGc = new int[101];
-        final int[] readsByGc = new int[101];
-        final long[] basesByGc = new long[101];
-        final long[] errorsByGc = new long[101];
-
-        final SamReader sam = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT);
-
-        if (!ASSUME_SORTED && sam.getFileHeader().getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
-            throw new PicardException("Header of input file " + INPUT.getAbsolutePath() + " indicates that it is not coordinate sorted.  " +
-                    "If you believe the records are in coordinate order, pass option ASSUME_SORTED=true.  If not, sort the file with SortSam.");
-        }
-        final PeekableIterator<SAMRecord> iterator = new PeekableIterator<SAMRecord>(sam.iterator());
-        final ReferenceSequenceFile referenceFile = ReferenceSequenceFileFactory.getReferenceSequenceFile(REFERENCE_SEQUENCE);
-
-        {
-            // Check that the sequence dictionaries match if present
-            final SAMSequenceDictionary referenceDictionary = referenceFile.getSequenceDictionary();
-            final SAMSequenceDictionary samFileDictionary = sam.getFileHeader().getSequenceDictionary();
-            if (referenceDictionary != null && samFileDictionary != null) {
-                SequenceUtil.assertSequenceDictionariesEqual(referenceDictionary, samFileDictionary);
+        save_header = header;
+    }
+    ////////////////////////////////////////////////////////////////////////////
+    // Loop over the reference and the reads and calculate the basic metrics
+    ////////////////////////////////////////////////////////////////////////////
+    @Override
+    protected void acceptRead(final SAMRecord rec, final ReferenceSequence ref) {
+        //if read is unaligned then ref is passed in as null
+        if (ref == null) {
+            if (!rec.getReadPairedFlag() || rec.getFirstOfPairFlag()) ++this.totalClusters;
+        } else {
+            //only do the recalculation of gc if current ref is different from last ref
+            if (ref.getContigIndex() != lastContig) {
+                refBases = ref.getBases();
+                StringUtil.toUpperCase(refBases);
```

Comment:
@mcovarr I think that this is a verbatim copy of the original code behavior. I agree that as written it is effectively a noop, but since @kbergin is not fixing bugs here we need the results to agree. It might make sense in a subsequent fix to change this, but right now I think she shouldn't change any functional aspect of the code. (my opinion)